core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/save/withSave.ts

```diff
@@ -0,0 +1,79 @@
+import { rxMethod } from '@ngrx/signals/rxjs-interop';
+import { EMPTY, pipe } from 'rxjs';
+
+import { inject } from '@angular/core';
+
+import { catchError, switchMap, tap } from 'rxjs/operators';
+
+import { DotPageApiResponse, DotPageApiService } from '../../../../services/dot-page-api.service';
+import { UVE_STATUS } from '../../../../shared/enums';
+import { PageContainer } from '../../../../shared/models';
+import { UVEState } from '../../../models';
+
+/**
+ * Add methods to save the page
+ *
+ * @export
+ * @return {*}
+ */
+export function withSave() {
+    return signalStoreFeature(
+        {
+            state: type<UVEState>()
+        },
+        withMethods((store) => {
+            const dotPageApiService = inject(DotPageApiService);
+
+            return {
+                savePage: rxMethod<PageContainer[]>(
+                    pipe(
+                        tap(() => {
+                            patchState(store, {
+                                $status: UVE_STATUS.LOADING
+                            });
+                        }),
+                        switchMap((pageContainers) => {
+                            const payload = {
+                                pageContainers,
+                                pageId: store.$pageAPIResponse().page.identifier,
+                                params: store.$params()
+                            };
+
+                            return dotPageApiService.save(payload).pipe(
+                                switchMap(() =>
+                                    dotPageApiService.get(payload.params).pipe(
+                                        tapResponse(
+                                            (pageAPIResponse: DotPageApiResponse) => {
+                                                patchState(store, {
+                                                    $status: UVE_STATUS.LOADED,
+                                                    $pageAPIResponse: pageAPIResponse
+                                                });
+                                            },
+                                            (e) => {
+                                                console.error(e);
+
+                                                patchState(store, {
+                                                    $status: UVE_STATUS.ERROR
+                                                });
+                                            }
+                                        )
+                                    )
+                                ),
+                                catchError((e) => {
+                                    console.error(e);
+                                    patchState(store, {
+                                        $status: UVE_STATUS.ERROR
+                                    });
+
+                                    return EMPTY;
+                                })
+                            );
```

Comment:
Can't we unify one catch?

Reply:
We cannot. Although it is doing the same, it shouldn't. One error should be only for the save request and the other for the get request. Right now we don't do anything, as we didn't do anything in the old store. We should create a ticket to discuss all errors and what are we going to do in them.